### PR TITLE
CoroutineUtils unit tests

### PR DESCRIPTION
**Pull request by oluiscabral**

### CODE REVIEW

File: lib/src/test/java/space/taran/arklib/utils/CoroutineUtilsTest.kt [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+import kotlin.system.measureTimeMillis
+
+class CoroutineUtilsTest {
+
+    @Test
+    fun testWithContextAndLock() = runBlocking {
+        val mutex = Mutex()
+        var sharedCounter = 0
+        val context = EmptyCoroutineContext
+        val concurrency = Random.nextInt(2, 5)
```
[the remark this answers is not in the document]
Comment (oluiscabral) [reply lib/src/test/java/space/taran/arklib/utils/CoroutineUtilsTest.kt:24]: Yes, I see no problem with it. Actually, I'll remove all those randomizers to replace them with a fixed values, so the tests always have reproducible values

File: lib/src/test/java/space/taran/arklib/utils/CoroutineUtilsTest.kt [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+                        delay(delayMillis)
+                        sharedCounter += 1
+                    }
+                }
+            }
+            jobs.forEach { it.join() }
+        }
+        println("Elapsed time: $elapsedTime ms")
+        assertEquals(concurrency, sharedCounter)
+        assertTrue(elapsedTime > concurrency * delayMillis)
```
[the remark this answers is not in the document]
Comment (oluiscabral) [reply lib/src/test/java/space/taran/arklib/utils/CoroutineUtilsTest.kt:39]: Yes

**Comment by mdrlzy** [comment]
Actually tests look great
But I'm not sure if they are needed
`tryUnlock` and `tickerFlow` don't seem to be used
`withContextAndLock` has very simple logic
@kirillt Do we need tests for dead code?

File: lib/src/test/java/space/taran/arklib/utils/CoroutineUtilsTest.kt [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        delay(initialDelayMillis)
+        assertEquals(0, collectedValues.size)
+        delay(delayMillis)
+        assertEquals(1, collectedValues.size)
+        delay(delayMillis * 2)
+        assertEquals(3, collectedValues.size)
+        job.cancelAndJoin()
+        delay(delayMillis * 2)
+        assertEquals(3, collectedValues.size)
+    }
```
[the remark this answers is not in the document]
Comment (mdrlzy) [reply lib/src/test/java/space/taran/arklib/utils/CoroutineUtilsTest.kt]: Everything is fine here

File: lib/build.gradle [hunk 4]
```diff
@@ -73,6 +73,9 @@ dependencies {
     kapt "com.github.bumptech.glide:compiler:4.11.0"
 
     implementation "org.jetbrains.kotlinx:kotlinx-coroutines-android:1.6.4"
+
+    testImplementation "junit:junit:4.13.2"
+    testImplementation "org.jetbrains.kotlinx:kotlinx-coroutines-test:1.7.3"
```
Comment (tanaytandon12) [inline lib/build.gradle:78]: @oluiscabral shouldn't the version of the kotlinx-coroutines-test library be the same as that of kotlinx-coroutines-android?

Comment (oluiscabral) [reply lib/build.gradle:78]: No, those are two independent packages/projects

**Review by kirillt — APPROVED** [review]
Looks good. Feel free to merge after resolving merge conflicts.

**Pull request merged**